ui/desktop/app/templates/scopes/scope/projects/targets/index.hbs

```diff
@@ -23,24 +23,51 @@
             aria-label={{t 'actions.search'}}
             {{on 'input' this.handleSearchInput}}
           />
-          <S.Dropdown as |DD|>
+          <S.Dropdown @onClose={{this.close}} as |DD|>
             <DD.ToggleButton
               @color='secondary'
               @text={{t 'resources.scope.title'}}
+              @count={{this.scopes.length}}
             />
             <DD.Header @hasDivider={{true}}>
               <Hds::Form::TextInput::Base
                 @type='search'
+                @value={{this.scopeSearchTerm}}
                 placeholder={{t 'actions.narrow-results'}}
                 aria-label={{t 'actions.narrow-results'}}
+                {{on 'search' this.filterItems}}
                 data-test-search
               />
             </DD.Header>
-            {{#each this.searchItems as |item|}}
-              <DD.Checkbox @value={{item}} data-test-checkbox={{item}}>
-                {{item}}
-              </DD.Checkbox>
-            {{/each}}
+            {{#unless this.availableScopes}}
+              <DD.Description @text={{t 'titles.no-results'}} />
+            {{/unless}}
+            {{#let (group-by this.availableScopes 'scopeModel') as |orgs|}}
```

Comment:
It's probably OK for scopes but I just want to mention we probably need to handle any resource that can be unbounded like the discussion of [targets](https://hashicorp.slack.com/archives/C0647GX0TLJ/p1701197068542169?thread_ts=1701196245.008619&cid=C0647GX0TLJ) in this thread initially.

We most likely need to cap the display just in case

Reply:
I actually take back that it's probably OK for scopes. I don't think we can ever assume what the user's usage will be and if it can be unbounded, we need to handle it.


https://github.com/hashicorp/boundary-ui/assets/5783847/b4b6a605-7d98-465f-8d04-240ab9a6b047

Reply:
This was discussed over in meeting and will be addressed in a later PR to limit scope of those changes.